log/log.go

```diff
@@ -30,18 +32,83 @@ var levelPrefix = [...]string{
 	LevelFatal:    "[FATAL] ",
 }
 
+var levelPriority = [...]syslog.Priority{
+	LevelDebug:    syslog.LOG_DEBUG,
+	LevelInfo:     syslog.LOG_INFO,
+	LevelWarning:  syslog.LOG_WARNING,
+	LevelError:    syslog.LOG_ERR,
+	LevelCritical: syslog.LOG_CRIT,
+	LevelFatal:    syslog.LOG_EMERG,
+}
+
 // Level stores the current logging level.
 var Level = LevelDebug
 
+var (
+	//SysLoggers maps each logging level to a *log.Logger that will be used for it.
+	SysLoggers = make([]*golog.Logger, LevelFatal+1)
+	//UseSyslog controls if syslog is used instead of stdout/err.
+	UseSyslog bool
+	//SyslogNetwork controls the network protocol syslog uses.
+	SyslogNetwork string
+	//SyslogRaddr is the address of a remote syslog instance.
+	SyslogRaddr string
+	//SyslogTag is the tag to pass to syslog.
+	SyslogTag string
+)
+
+func init() {
+	flag.IntVar(&Level, "loglevel", LevelDebug, "Log level (0 = DEBUG, 5 = FATAL)")
+	flag.BoolVar(&UseSyslog, "syslog", false, "Whether or not to use syslog logging")
+	flag.StringVar(&SyslogNetwork, "syslog-network", "udp", "Syslog network to use.")
+	flag.StringVar(&SyslogRaddr, "syslog-remote", "", "Syslog server to use. Defaults to local on empty string.")
+	flag.StringVar(&SyslogTag, "syslog-tag", "udp", "Syslog tag to use.")
+}
+
+func getSysLogger(l int) *golog.Logger {
+	if SysLoggers[l] != nil {
+		return SysLoggers[l]
+	}
+
+	w, err := syslog.Dial(SyslogNetwork, SyslogRaddr, levelPriority[l], SyslogTag)
+	if err == nil {
+		SysLoggers[l] = golog.New(w, levelPrefix[l], 0)
+		return SysLoggers[l]
+	}
+
+	golog.Println("Unable to dial syslog:", err)
+	return nil
+}
+
 func outputf(l int, format string, v []interface{}) {
 	if l >= Level {
-		golog.Printf(fmt.Sprint(levelPrefix[l], format), v...)
+		msg := fmt.Sprintf(fmt.Sprint(levelPrefix[l], format), v...)
+		if UseSyslog {
+			logger := getSysLogger(l)
+			if logger != nil {
+				logger.Print(msg)
+			} else {
+				golog.Print(msg)
+			}
+		} else {
+			golog.Print(msg)
+		}
 	}
 }
 
 func output(l int, v []interface{}) {
 	if l >= Level {
-		golog.Print(levelPrefix[l], fmt.Sprint(v...))
+		msg := fmt.Sprint(levelPrefix[l], fmt.Sprint(v...))
+		if UseSyslog {
```

Comment:
This could just be
```Go
if UseSyslog {
	if logger := getSysLogger(l); logger != nil {
		logger.Print(msg)
		return
	}
}
golog.Print(msg)	
```

And it should probably be factored out into a helper function that both `output` and `outputf` call.